lib/Fhp/Action/SendSEPATransfer.php

```diff
@@ -35,21 +47,52 @@ class SendSEPATransfer extends BaseAction
      */
     public static function create(SEPAAccount $account, string $painMessage): SendSEPATransfer
     {
-        if (preg_match('/xmlns="(.*?)"/', $painMessage, $match) === false) {
+        if (preg_match('/xmlns="(.*?)"/', $painMessage, $xmlns) === false) {
             throw new \InvalidArgumentException('xmlns not found in the PAIN message');
         }
+
+        // Check whether the PAIN message contains the correct message root
+        if (strstr($painMessage, '<CstmrCdtTrfInitn>') === false) {
```

Comment:
!str_contains

Reply:
str_contains is PHP 8 only. I think phpFinTs retains compatibility with PHP 7 for now? Anyway, we would need that compatibility on our side (for now).

Reply:
Right, sorry, please ignore.